packages/teleport/src/components/MfaDeviceList/MfaDeviceList.tsx

```diff
@@ -0,0 +1,150 @@
+import React, { useState, useMemo } from 'react';
+import styled from 'styled-components';
+import { sortBy } from 'lodash';
+import { ButtonBorder, Text } from 'design';
+import {
+  Column,
+  SortHeaderCell,
+  Cell,
+  TextCell,
+  SortTypes,
+  Table,
+} from 'design/DataTable';
+import { displayDate } from 'shared/services/loc';
+import { MfaDevice } from 'teleport/services/mfa/types';
+
+export default function MfaDeviceList({
+  devices = [],
+  onDelete,
+  isRecoveryFlow,
+}: Props) {
+  const [sortDir, setSortDir] = useState<Record<string, string>>({
+    registeredDate: SortTypes.ASC,
+  });
+
+  const mostRecentDevice = useMemo(() => sort(devices)[0], []);
+
+  function sort(data) {
+    const columnKey = Object.getOwnPropertyNames(sortDir)[0];
+    const sorted = sortBy(data, columnKey);
+    if (sortDir[columnKey] === SortTypes.ASC) {
+      return sorted.reverse();
+    }
+
+    return sorted;
+  }
+
+  function onSortChange(columnKey: string, sortDir: string) {
+    setSortDir({ [columnKey]: sortDir });
+  }
+
+  const data = sort(devices);
+
+  return (
+    <StyledTable
+      data={data}
+      style={isRecoveryFlow ? { overflow: 'hidden', borderRadius: '8px' } : {}}
+    >
```

Comment:
instead of `isRecoveryFlow`, you can propagate the props to the table

```tsx
const {
  devices = [],
  onDelete,
 ...styles
}: Props)

return (
<StyledTable data={data} ...styles>
```    

this way, a consumer of this component can apply custom styles

```tsx
<MfaDeviceList style={...} />
```